ambry-api/src/main/java/com/github/ambry/quota/QuotaChargeCallback.java

```diff
@@ -51,22 +50,23 @@ public void charge(long chunkSize) throws RouterException {
           if (throttlingRecommendation != null && throttlingRecommendation.shouldThrottle() && shouldThrottle) {
```

Comment:
also, I think the precondition to check if we should throttle is getting out of control, we have throttling recommendation, and should throttle parameter, and quota mode and throttleInProgressRequests, at some point, we should figure out a way to simplify all these.